polyfill/intersectionobserver-polyfill.js

```diff
@@ -0,0 +1,188 @@
+(function() {
+  // TODO: rootMargin are completely ignored for now
+  
+  // Constructor
+  window.IntersectionObserver = function(callback, options) {
+    options = options || {};
+    
+    if(!(callback instanceof Function)) {
+      throw('callback needs to be a function');
+    }
+    
+    this._callback = callback;
+    this._root = options.root || null;
```

Comment:
All `observe()` calls will fail if `root` is invalid. So this check is not strictly needed, but I agree it’s probably a good idea to add that check.